src/bitter/bit_field.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Steinwurf ApS 2016.
+// All Rights Reserved
+//
+// Distributed under the "BSD License". See the accompanying LICENSE.rst file.
+#pragma once
+
+#include "size_in_bits.hpp"
+
+#include <type_traits>
+#include <cstdint>
+#include <cassert>
+
+namespace bitter
+{
+template<typename DataType>
+class bit_field
+{
+/// @brief Small class used for wrapping a bitfield
+/// @param value is the value of the bit we are wrapping
+/// @param size is the size of the field in bits
+public:
+    bit_field(DataType value, uint32_t size) :
+        m_value(value),
+        m_size(size)
+    {
+
+    }
+
+
+    /// @return m_value cast to the type of ReturnType
+    template<typename ReturnType>
+    ReturnType read_as()
+    {
+        // Check if the field fits into the size of ReturnType
+        assert(sizeof(ReturnType) <= sizeof(DataType));
```

Comment:
This check is wrong, but a check should be there. What it should check is that `ReturnType` has at least as many bits as field value the user is trying access.